packages/sanity/src/datastores/authStore/createAuthStore.ts

```diff
@@ -84,12 +84,6 @@ const saveToken = ({token, projectId}: {token: string; projectId: string}): void
 }
 
 const getCurrentUser = async (client: SanityClient) => {
```

Comment:
Nice optimization! 👍 

But the CORS error screen is not displayed after these changes. I investigated it a bit and I think these changes will make it work. Let me know what you think!

```suggestion
const getCurrentUser = async (client: SanityClient) => {
  try {
    const user = await client
      .request({
        uri: '/users/me',
        withCredentials: true,
        tag: 'users.get-current',
      })
      .then((res) => res)
      .catch((err) => {
        // 401 means the user had some kind of credentials, but failed to authenticate,
        // we should clear any local token in this case and treat it as if the used was
        // logged out
        if (err.statusCode === 401) {
          clearToken(client.config().projectId || '')
        }
        return null
      })

    // Request failed for some other reason, see if this was a CORS-error by
    // checking the `/ping` endpoint, which allows all origins
    if (!user) {
      const validCorsConfig = await client
        .request({uri: '/ping', withCredentials: false, tag: 'cors-check'})
        .then(() => true)

      // Throw a specific error on CORS-errors, to allow us to show a customized dialog
      if (validCorsConfig) {
        throw new CorsOriginError({projectId: client.config()?.projectId})
      }
    }

    // If the user came back with an id, assume it's a full CurrentUser
    return typeof user?.id === 'string' ? user : null
  } catch (err) {
    // Some non-CORS error, re-throw
    throw err
  }
}
```